src/sentry/integrations/pagerduty/notify_action.py

```diff
@@ -0,0 +1,147 @@
+    service = forms.ChoiceField(required=False, choices=(), widget=forms.Select())
+
+    def __init__(self, *args, **kwargs):
+        integrations = [(i.id, i.name) for i in kwargs.pop("integrations")]
+        services = kwargs.pop("services")
+
+        super(PagerDutyNotifyServiceForm, self).__init__(*args, **kwargs)
+        if integrations:
+            self.fields["account"].initial = integrations[0][0]
+
+        self.fields["account"].choices = integrations
+        self.fields["account"].widget.choices = self.fields["account"].choices
+
+        try:
+            service_project = PagerDutyServiceProject.objects.get(
+                project_id=self.project.id,
+            )
+        except PagerDutyServiceProject.DoesNotExist:
+            service_project = None
+
+        if services:
+            self.fields["service"].initial = service_project.pagerduty_service_id or services[0][0]
+
+        self.fields["service"].choices = services
+        self.fields["service"].widget.choices = self.fields["service"].choices
+
+
+class PagerDutyNotifyServiceAction(EventAction):
+    form_cls = PagerDutyNotifyServiceForm
+    label = "Send a notification to PagerDuty account {account} and service {service}"
+
+    def __init__(self, *args, **kwargs):
+        super(PagerDutyNotifyServiceAction, self).__init__(*args, **kwargs)
+        self.form_fields = {
+            "account": {
+                "type": "choice",
+                "choices": [(i.id, i.name) for i in self.get_integrations()],
+            },
+            "service": {
+                "type": "choice",
+                "choices": self.get_services(),
+            },
+
+        }
+        initial_service = self.get_initial_service()
+        if initial_service:
+            self.form_fields["service"]["initial"] = initial_service
+
+    def get_initial_service(self):
+        try:
+            # service_id here is the id of the PagerDutyService record, not the
+            # service_id that is generated by pagerduty
+            service_id = PagerDutyServiceProject.objects.get(
+                project=self.project,
+            ).pagerduty_service_id
+        except PagerDutyServiceProject.DoesNotExist:
+            service_id = None
+
+        return service_id
+
+    def after(self, event, state):
+        try:
+            integration = Integration.objects.get(
+                provider="pagerduty",
+                organizations=self.project.organization,
+                status=ObjectStatus.VISIBLE,
+            )
+        except Integration.DoesNotExist:
+            # integration removed but rule still exists
+            return
+
+        try:
+            service = PagerDutyService.objects.get(pk=self.get_option("service"))
+        except PagerDutyService.DoesNotExist:
+            return
+
+        def send_notification(event, futures):
+            client = PagerDutyClient(integration_key=service.integration_key)
+            client.send_trigger(event)
```

Comment:
What happens when this request fails?

Reply:
hmm we should probably log this. Ideally we could use it to see if a service has been delete but from testing it out we still get a 202 even if the service has been deleted

Reply:
I was more thinking about if/when pagerduty is having an outage.